### PR TITLE
fix(ui): add tokens for ShowPassword button error state 

### PR DESCRIPTION
**Pull request by esauerbo**
#### Issue #, if available

#4410 

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by thaddmt — APPROVED** [review]

**Review by ioanabrooks — APPROVED** [review]

File: packages/ui/src/theme/tokens/components/passwordField.ts [hunk 1]
```diff
@@ -37,6 +38,15 @@ export const passwordfield: Required<PasswordFieldTokens<'default'>> = {
       },
       color: { value: '{components.button._disabled.color.value}' },
     },
+    _error: {
```
Comment (hbuchel) [inline packages/ui/src/theme/tokens/components/passwordField.ts:41]: Can we add the other interaction states? If we allow theming of the base error styles for this button, we should allow developers to theme _focus, _hover, _active? So then 
```
_error?: StateTokens<Output>;
```
would become something like:
```
_error?: ErrorStateTokens<Output>;
```
and `ErrorStateTokens` would be something like:

```
type ErrorStateTokens<Output> = DesignTokenProperties<
  'backgroundColor' | 'borderColor' | 'color',
  Output
> & {
  _hover?: StateTokens<Output>;
  _focus?: StateWithShadowTokens<Output>;
  _active?: StateTokens<Output>;
};
```
Basically mimicking how we build the [button token types for a color theme](https://github.com/aws-amplify/amplify-ui/blob/main/packages/ui/src/theme/tokens/components/button.ts#L8). And then you can add the _hover, _focus, _active to this _error: {} definition.

**Review by hbuchel — COMMENTED** [review]
Thanks @esauerbo! I left a comment about the missing interaction state tokens.

File: packages/ui/src/theme/tokens/components/passwordField.ts [hunk 2]
```diff
@@ -37,6 +50,54 @@ export const passwordfield: Required<PasswordFieldTokens<'default'>> = {
...
+      color: { value: '{components.button.outlined.error.color.value}' },
+      backgroundColor: {
+        value: '{components.button.outlined.error.backgroundColor.value}',
+      },
+      borderColor: {
+        value: '{components.button.outlined.error.backgroundColor.value}',
```
Comment (hbuchel) [inline packages/ui/src/theme/tokens/components/passwordField.ts:59]:
```suggestion
        value: '{components.button.outlined.error.borderColor.value}',
```

Is this correct? Looks like there might be a mismatch between borderColor property using a backgroundColor value

Comment (esauerbo) [reply packages/ui/src/theme/tokens/components/passwordField.ts:59]: Good catch

File: packages/ui/src/theme/css/component/passwordField.scss
Comment (hbuchel) [inline packages/ui/src/theme/css/component/passwordField.scss:1]: This bit can be removed and should fix the theming examples not working for some states.

```
.amplify-field__show-password {
  &--error {
    color: var(--amplify-components-fieldcontrol-error-color);
    border-color: var(--amplify-components-fieldcontrol-error-border-color);
  }
}
```

**Review by thaddmt — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]
LGTM! Thank you for the fixes/updates!

**Pull request merged**